### DISCUSSION TITLE
[Rails] Let's make it possible to set the available i18n languages per community

### DISCUSSION THREAD

**Discussion by rudokemper** [Terrastories/terrastories#765] [Temporarily Archived Tickets]
Currently, when you open Terrastories (i.e. by going to http://our.terrastories.io/), there are a defined number of i18n languages that show up. The way this currently works is that in `application.rb` there is some code that automatically loads in whatever languages are defined in `config/locales/`. This has worked for now, but will get unsustainable as more languages get added to Terrastories.

Let's (1) specify a default set of languages that show up for a non-logged in user, and (2) allow community administrators to set which languages they want to show for their community when logged in, and the default selected language for their community.

See acceptance criteria below.

## Acceptance criteria:

- [ ] For a non-logged in user loading the Terrastories welcome page, instead of populating all of the languages in `config/locales` into the language picker, let's only provide these: `en`, `es`, `ja`, `mat`, `pt`. (We can manually add more default languages in the future if those translations are provided).
- [ ] The default selected language should be `en` (which is currently the case already). 
- [ ] A community administrator is able to see a field called "Languages" which is a select-multiple has_many field (akin to how Stories show on the Speaker dashboard, for example) showing all of the languages available in `config/locales/`. Using this field, community administrator is able to select the subset of languages they want to show for the community welcome screen.
- [ ] There should be a "Default Language" field, which is a typeahead pulling values from the available languages. Using this field the administrator can set any of the languages in Terrastories as the default for logged in users.
- [ ] For a logged in user loading the Terrastories welcome page, they should see only the languages that have been selected by the community administrator.
- [ ] The default language picker options for a community should also be `en`, `es`, `ja`, `mat`, `pt`, until overridden by the community administrator.
- [ ] The default selected language for a community should be `en`.
- [ ] When a user logs out of the community, they should return to seeing the five default languages for a non-logged in user as per the first step.

## Assets
Screenshot of the welcome screen box with the language picker at the bottom. Currently `nl` is included as a default; it should not be.
![image](https://user-images.githubusercontent.com/31662219/140209954-36fecea9-8532-4697-958e-7a3cf3be9898.png)

**Comment by arlohal** [comment]
Hello! I'm looking for something on this project to dive into and was wondering if this issue was still relevant and something I could potentially start working on. If not, is there another issue that is more pressing?

**Comment by rudokemper** [comment]
Hi @arlohal! Thank you for your interest :) We are actually about to undergo a major refactor that is going to touch most issues, including this one. So it's not a great time to work on this ticket. There may be others that I can share with you, but it will take me a few days to do so. Post-refactor, this issue will definitely be resurfaced too!
